### PR TITLE
Add Elasticsearch options

### PR DESCRIPTION
**Pull request by dotpy3**
This PR adds new flags related to Elasticsearch logging:

+ `--elasticsearch-prefix` - Adds a user-defined prefix to the Elasticsearch index where documents are stored.
For the moment, they are stored in the `<component>-<date>` index (e.g. `handler-2017.07.10`). If `--elasticsearch-prefix` is set, they would be stored in `<prefix>-<component>-<date>`(e.g. with `europe` prefix, `europe-handler-2017.07.10`).
+ `--elasticsearch-username` and `--elasticsearch-password` - Allow to provide credentials if the Elasticsearch instance requires authentication.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -22,10 +22,10 @@ import (
 	jsonHandler "github.com/apex/log/handlers/json"
 	levelHandler "github.com/apex/log/handlers/level"
 	multiHandler "github.com/apex/log/handlers/multi"
+	"github.com/dotpy3/go-elastic"
```
Comment (htdvisser) [inline cmd/root.go]: I don't really like this. Can't we wait for the PR to be merged? In the meantime, we can also just vendor from your fork.

File: cmd/root.go [hunk 2]
```diff
@@ -72,9 +72,21 @@ var RootCmd = &cobra.Command{
 			esClient.HTTPClient = &http.Client{
 				Timeout: 5 * time.Second,
 			}
+
+			username := viper.GetString("elasticsearch-username")
```
Comment (htdvisser) [inline cmd/root.go]: Let's use `esUsername`/`esPassword`, otherwise it's too generic

File: cmd/root.go [hunk 3]
```diff
@@ -72,9 +72,21 @@ var RootCmd = &cobra.Command{
...
+			password := viper.GetString("elasticsearch-password")
+			if username != "" {
+				esClient.SetAuthCredentials(username, password)
+			}
+
+			indexPrefix := cmd.Name()
```
Comment (htdvisser) [inline cmd/root.go]: then also `esPrefix`

**Comment by dotpy3** [comment]
Let's wait for the `tj/go-elastic` PR status to change.

**Comment by htdvisser** [comment]
Blocked by https://github.com/tj/go-elastic/pull/3

**Pull request merged**